src/potion.c

```diff
@@ -540,6 +540,43 @@ make_glib(int xtime)
         update_inventory();
 }
 
+void
+make_withering(long xtime, boolean talk)
+{
+    long old = (HWithering & TIMEOUT);
+    boolean was_withering = Withering, no_longer_withering;
+
+    if (Unaware)
+        talk = FALSE;
+
+    set_itimeout(&HWithering, xtime);
+    no_longer_withering = !Withering;
+    if ((xtime != 0L) ^ (old != 0L)) {
+        g.context.botl = TRUE;
+        if (talk) {
+            if (old) {
+                if (no_longer_withering) {
+                    You("stop withering.");
+                }
+                else {
+                    Your("withering slows, but only briefly.");
```

Comment:
make_withering is not *currently* used to increase the amount of withering but that's not a hard rule, right? Shouldn't this also account for if timing-out withering is now greater than old?

Reply:
It should already handle the "preexisting withering timeout is increased" (which can happen via `#wizintrinsics` in this patch, if you select withering from the menu when already withering) on line 572.

IIRC the only case where it doesn't print a message is if you're already withering from a non-timeout source, and a withering timeout is started, but I'm not 100% sure about that.